src/librustc/mir/visit.rs

```diff
@@ -734,6 +737,12 @@ macro_rules! make_mir_visitor {
 make_mir_visitor!(Visitor,);
 make_mir_visitor!(MutVisitor,mut);
 
+pub enum PositionalInfo {
```

Comment:
This is the one part I wasn't sure about. I could have just passed `Span`s as the third parameter to `visit_ty`, but this leaves the visit implementations a little flexible for future uses. Also, is this the best place for this enum?